Modules/_operator.c

```diff
@@ -1700,19 +1700,25 @@ methodcaller_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
     _PyUnicode_InternMortal(interp, &name);
     mc->name = name;
 
-    mc->xargs = Py_XNewRef(args); // allows us to use borrowed references
+    mc->args = PyTuple_GetSlice(args, 1, PyTuple_GET_SIZE(args));
```

Comment:
`PyObject_GC_New` doesn't doesn't initialize the contents, so you need to make sure that everything is initialized (possibly to NULL) before anything that might call `Py_DECREF(mc)`.

For example, you can move the assignments to `mc->kwds` ... `mc->vectorcall_kwnames` up above the `mc->args` assignment, which may possibly fail.

Reply:
Nice catch!